gwctl/pkg/printer/backends.go

```diff
@@ -28,7 +34,96 @@ import (
 )
 
 type BackendsPrinter struct {
-	Out io.Writer
+	Out   io.Writer
+	Clock clock.Clock
+}
+
+func (bp *BackendsPrinter) Print(resourceModel *resourcediscovery.ResourceModel) {
+	tw := tabwriter.NewWriter(bp.Out, 0, 0, 2, ' ', 0)
+	row := []string{"NAMESPACE", "NAME", "TYPE", "REFERRED BY ROUTES", "AGE", "POLICIES"}
+	tw.Write([]byte(strings.Join(row, "\t") + "\n"))
+
+	backendNodes := make([]*resourcediscovery.BackendNode, 0, len(resourceModel.Backends))
+	for _, backendNode := range resourceModel.Backends {
+		backendNodes = append(backendNodes, backendNode)
+	}
+
+	sort.Slice(backendNodes, func(i, j int) bool {
+		if backendNodes[i].Backend.GetNamespace() != backendNodes[j].Backend.GetNamespace() {
+			return backendNodes[i].Backend.GetNamespace() < backendNodes[j].Backend.GetNamespace()
+		}
+		return backendNodes[i].Backend.GetName() < backendNodes[j].Backend.GetName()
+	})
+
+	httpRouteNodes := make([]*resourcediscovery.HTTPRouteNode, 0, len(resourceModel.HTTPRoutes))
+	for _, httpRouteNode := range resourceModel.HTTPRoutes {
+		httpRouteNodes = append(httpRouteNodes, httpRouteNode)
+	}
+
+	for _, backendNode := range backendNodes {
+		backend := backendNode.Backend
+
+		parentHttpRoutes := []string{}
+		remainderHttpRoutes := 0
+		associatedDistinctPolicies := map[string]any{}
+
+		// the nested loops could've been made more efficient (on textbook) by using map between httpRouteNodes and backendRefs
+		// and leveraging the O(1) lookups but that seemed to sacrifice the code readability/structure, so didn't choose to do so
+		for _, httpRouteNode := range httpRouteNodes {
```

Comment:
Did we perhaps overlook this or is there some reason to not use the mappings already provided through the resourceModel (i.e. using the `BackendNode.HTTPRoutes`? If there's a bug in those mappings, we should try and fix this (separately, if required)

https://github.com/kubernetes-sigs/gateway-api/blob/29e68bffffb9af568e35545305d78d0001a1a0f7/gwctl/pkg/resourcediscovery/nodes.go#L215-L228